trace-utils/src/test_utils/datadog_test_agent.rs

```diff
@@ -405,4 +432,78 @@ impl DatadogTestAgent {
             res.status()
         );
     }
+
+    /// Sends an HTTP request to the Datadog Test Agent with rudimentary retry logic.
+    ///
+    /// In rare situations when tests are running on CI, the container running the test agent may
+    /// reset the network connection even after ready states pass. Instead of adding arbitrary
+    /// sleeps to these tests, we can just retry the request. This function should not be used for
+    /// requests that are actually being tested, like sending payloads to the test agent. It should
+    /// only be used for requests to setup the test. Examples of when you would use this
+    /// function are for starting sessions or getting snapshot results.
+    ///
+    /// # Arguments
+    ///
+    /// * `req` - A `Request<Body>` representing the HTTP request to be sent.
+    /// * `max_retries` - An `i32` specifying the maximum number of retry attempts.
```

Comment:
The name of the parameter also needs to be updated here

Reply:
good catch, thank you.